src/contracts/StakingChef.sol

```diff
@@ -0,0 +1,158 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "./Chef.sol";
+import "./MetricToken.sol";
+
+//TODO StakeMetric, updateStaker, removeStaker, updateAccumulatedStakingRewards, harvest and Claim can be moved to Chef
+
+contract StakingChef is Chef {
+    using SafeMath for uint256;
+    Staker[] private _stakes;
+    uint256 private _totalAllocPoint;
```

Comment:
I wonder if these two can be in Chef or we create functions in Chef for getters and setters which can be overiden